web/beacon-app/src/application/config/react-query.tsx

```diff
@@ -4,15 +4,15 @@ const queryClient = new QueryClient({
   // define default options for all queries
   defaultOptions: {
     queries: {
-      staleTime: 0,
+      staleTime: Infinity,
```

Comment:
I think here you are disabling the caching feature by default. May I know why ?

Reply:
No, it does not disable the caching feature. React query has 2 interesting config properties cacheTime and StateTime.
- The cacheTime indicates how long a response should be cached before it is discarded. as we know the Cache time is set to 5 minutes by default
- StaleTime indicates how long a response is considered fresh after it has been stale for a certain period.

Our request does not need to be fresh in our case. The InvalidQuery method is used on mutation to request fresh data for that particular endpoint that why i set stateTime to infinity

Reply:
A staleTime of 0 means that when a request is made, react query will serve the cached data first, then make a background request and populate the new data if it differs from the cached data.